### ISSUE TITLE
Update to 1.17.4

### ISSUE THREAD

**Issue by Perkolator** [flathub/com.makemkv.MakeMKV#53]
Hello, please update to 1.17.4 (29.5.2023), the previous version 1.17.3 expires today. Many thanks!

**Comment by Procsiab** [comment]
Hi, I have the same issue and it seems that the pull request https://github.com/flathub/com.makemkv.MakeMKV/pull/55 should be able to fix this, by upgrading the software.
However the automatic build for the Flatpak seems to be failing because of the expired TLS certificate on MakeMKV's website, which in turn make some check fail during the build (as shown on the logs [here](https://buildbot.flathub.org/#/builders/19/builds/5464/steps/7/logs/stdio))

The error can be triggered even from a browser (tested with the latest Firefox at the time of writing): by going to the URL https://makemkv.com the following error is reported:
```
SEC_ERROR_EXPIRED_CERTIFICATE
```
In addition, from the browser it is possible to easily inspect the certificate and confirm that its "Not After" property is set to `Sat, 06 May 2023 00:14:31 GMT`

Back to the Flatpak build: the failing step involves the run of the tool `org.flathub.flatpak-external-data-checker`; I tryed to replicate the build on my local machine to see if it would succeed without running that checker tool: I got the error discussed in the following thread on MakeMKV's forums: https://forum.makemkv.com/forum/viewtopic.php?f=3&p=135873
This seems to be related to the very recent FFMpeg 6.0, which deprecates the `av_mallocz_array` function; I followed the advice inside the forum thread and replaced it with `av_calloc` inside the file `libffabi/src/ffabi.c` at line `481` for the module `makemkv-oss`.
To use the changed code, after downloading the TAR.GZ archive of the `makemkv-oss` code, extracting, applying the patch to the aforementioned file and finally repacking it as TAR.GZ, I also patched the YAML manifest file for the module `makemkv-oss`, updating the line with the correct SHA256 and changed from `url` to `path` with the value of the absolute path of that archive on my machine. 

I could finally let the build run for on my Intel Core i3 5005U, which took about half an hour. By the way, the build succeded! I can also report that the MakeMKV Flatpak is functional at version 1.17.4.
![gui_license_makemkv-1 17 4](https://github.com/flathub/com.makemkv.MakeMKV/assets/29021670/05204788-2f57-4b87-a561-592a95b81c71)

**Comment by JamesBelchamber** [comment]
Build is in #56, please test :)

**Comment by Procsiab** [comment]
Before testing, I looked at the new commits from @JamesBelchamber to understand what was needed to make the build succeed:
- the source code download URL was changed adding `www.` before the website's name: it appears to not trigger the certificate error but a redirect to the "HTTP version" of the download page; I am still not sure if this is a permanent fix for the expired certificate.
A test with `curl` towards the _working_ URL:
```bash
$ curl -I https://www.makemkv.com/download
HTTP/2 301 
date: Sat, 03 Jun 2023 16:47:53 GMT
content-type: text/html; charset=iso-8859-1
location: http://www.makemkv.com/download/
...
```
- The FFMpeg version was rolled back to `5.1.3`, to avoid editing the codebase. In that regard, it seems the only option since it does not apper to be an easy way to send the MakeMKV devs a "pull request"...

I can confirm that the test build is working on my machine ™

![Screenshot from 2023-06-03 18-55-56](https://github.com/flathub/com.makemkv.MakeMKV/assets/29021670/01c48b4b-79c6-428e-bff4-9dfe88151dd7)

**Comment by Perkolator** [comment]
> it does not apper to be an easy way to send the MakeMKV devs a "pull request"

If you want to try, `support@makemkv.com` was previously in the FAQ section of the site (now that section doesn't work).

**Comment by Procsiab** [comment]
I tried to send them a mail, referencing also that forum thread with the discussion on FFMpeg 6 support; probably it should be better to follow on on this matter into another issue, if they reply.

**Comment by michaelkuhn** [comment]
Fixed by #56.
